src/journal.rs

```diff
@@ -325,13 +326,25 @@ impl Journal {
         Ok(cursor)
     }
 
-    /// Returns timestamp at which current journal is recorded
+    /// Returns timestamp at which current journal entry is recorded.
     pub fn timestamp(&self) -> Result<time::SystemTime> {
         let mut timestamp_us: u64 = 0;
         sd_try!(ffi::sd_journal_get_realtime_usec(self.j, &mut timestamp_us));
         Ok(system_time_from_realtime_usec(timestamp_us))
     }
 
+    /// Returns monotonic timestamp and boot ID at which current journal entry is recorded.
+    pub fn monotonic_timestamp(&self) -> Result<(u64, Id128)> {
+        let mut monotonic_timestamp_us: u64 = 0;
+        let mut id: sd_id128_t = unsafe { uninitialized() };
```

Comment:
I think this is unidiomatic and you can't actually prove that the uninitialized memory is not leaked out (that depends on `ffi::sd_journal_get_monotonic_usec` internals).
I'd suggest to implement the equivalent of `SD_ID128_NULL` (see `sd-id128(3)`) on `Id128` and use that instead for the initialization here.

Reply:
This pattern is lifted from `id128.rs`, though I agree we should use something like `SD_ID128_NULL`. Mind if I do that in a follow-up?

> that depends on `ffi::sd_journal_get_monotonic_usec` internals

Interesting, can you expand on that? If the call fails, then the value gets dropped right? Also note it's allocated on the stack.